pulser/sequence.py

```diff
@@ -638,39 +657,28 @@ def add(
                     if op.targets & last.targets or protocol == "wait-for-all":
                         current_max_t = op.tf
                         break
-        ti = current_max_t
-        if ti > t0:
-            # Insert a delay
-            delay_duration = ti - t0
 
+        delay_duration = current_max_t - t0
+        # Find last pulse and compare phase
+        for op in self._schedule[channel][::-1]:
+            if isinstance(op.type, Pulse):
+                if op.type.phase != pulse.phase:
+                    delay_duration = max(
+                        delay_duration, self._channels[channel].phase_jump_time
+                    )
+                break
+
+        if delay_duration > 0:
```

Comment:
It can't be `< 0` because `delay_duration <= current_max_t - t0` and `current_max_t => t0` by definition. If `== 0` then you'll simply end up with `ti = t0`; `min_duration` only needs to be respected if there is a delay.